package/MDAnalysis/lib/formats/libmdaxdr.pyx

```diff
@@ -318,7 +322,10 @@ cdef class _XDRFile:
         set_offsets
         """
         if not self._has_offsets:
+            # wrap this in try/except to ensure it gets closed?
+            self.open(self.fname, self._mode)
```

Comment:
why not 

```python
with self:
    ...
```

that might work and should close the file when an IOError is thrown